pyxrf/model/fit_spectrum.py

```diff
@@ -1834,7 +1836,33 @@ def fit_pixel_multiprocess_nnls(exp_data, matv, param,
     num_processors_to_use = multiprocessing.cpu_count()
 
     logger.info('cpu count: {}'.format(num_processors_to_use))
-    pool = multiprocessing.Pool(num_processors_to_use)
+
+    # TODO: find better permanent solution for issue with multiprocessing on Catalina MacOS
+    # =======================================================================================
+    # The following is a temporary patch for the issue with multiprocessing on Catalina MacOS.
+    # The issue is solved by disabling multiprocessing if current OS is Catalina MacOS.
+    # The code will change when the better solution is found.
+
+    # Check if the OS is MacOS Catalina (10.15) or older
+    disable_multiprocessing = False
+    try:
+        os_version = platform.mac_ver()[0]
+        # 'os_version' is an empty string if the system is not MacOS
+        if os_version:  # os_version has format '10.15.3'
+            ver = os_version.split('.')
+            if int(ver[0]) >= 10 and int(ver[1]) >= 15:  # Catalina MacOS is 10.15
```

Comment:
Why not to check it with the `LooseVersion`?

Reply:
Done

Reply:
Thank you.